### ISSUE TITLE
droplet/S3 does not support TLS1.3 and/or ECDSA

### ISSUE THREAD

**Issue by lluuaapp** [bareos/bareos#2036]
### Bareos component version

All components: version 23.1.0 set

### Steps to reproduce

Configure an S3 backend with an ECDSA/ECC certificate and only TLS 1.3 support (e.g. Hetzner Object Storage offers free tiers while it's in beta until the end of the month). My configuration is:

```
backend = s3
use_https = true
cert_verif = true/false
aws_auth_sign_version = 4
access_key = <key>
secret_key = <secret>
aws_region = fsn1
host = fsn1.your-objectstorage.com
pricing_dir = ""
```

Start bconsole and execute status storage=<name>. If you have cert_verif enabled, it can not verify the certificate:

`init_ssl_conn:383: SSL certificate verification status: 18: self-signed certificate`

If you disable the verification, the connection later fails to read the answer:

`dpl_read_http_reply_buffered:459: read line: DPL_EIO`

You can enable the droplet traces by starting the storage with the following environent var set:

`DPL_TRACE_LEVEL=65535`

Libdroplet works with another S3 storage I am using, also with an Let's Encrypt certificate. But this one uses CN = R10 (RSA-SHA256), not CN = E6 (ecdsa-with-SHA384) like the failing one.

### Expected results

The storage daemon can connect to the S3 host via droplet.

### Actual results

The connection (or the verification) fails.

### Environment

```markdown
- OS: `Debian 12.8, Bookworm`
- component: Bareos Storage Daemon, lib droplet plugin
```

### Relevant log output

```shell
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/backend/s3/reqbuilder.c:dpl_s3_req_build:253: req_build method=HEAD bucket=testbucket resource=bareos-test subresource=(null)
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [4] ./core/src/droplet/libdroplet/src/conn.c:conn_open:407: conn_open 88.198.120.64:443
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [4] ./core/src/droplet/libdroplet/src/conn.c:conn_open:488: conn_open conn=0x7fe764207ff0
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/httprequest.c:dpl_req_gen_http_request:226: req_gen_http_request resource=bareos-test
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/httprequest.c:dpl_req_gen_http_request:307: header='Connection' value='Keep-Alive'
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/httprequest.c:dpl_req_gen_http_request:307: header='Authorization' value='AWS key:secret='
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/httprequest.c:dpl_req_gen_http_request:307: header='Host' value='testbucket.fsn1.your-objectstorage.com:443'
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [40] ./core/src/droplet/libdroplet/src/httprequest.c:dpl_req_gen_http_request:307: header='Date' value='Sat, 23 Nov 2024 11:48:08 GMT'
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [8] ./core/src/droplet/libdroplet/src/conn.c:dpl_conn_writev_all:864: writev conn=0x7fe764207ff0 https=1 size=222
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [8] ./core/src/droplet/libdroplet/src/httpreply.c:read_line:172: read conn=0x7fe764207ff0 https=1 size=8192
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [1] ./core/src/droplet/libdroplet/src/httpreply.c:dpl_read_http_reply_buffered:459: read line: DPL_EIO
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [4] ./core/src/droplet/libdroplet/src/conn.c:dpl_blacklist_host:546: blacklisting testbucket.fsn1.your-objectstorage.com:443
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [4] ./core/src/droplet/libdroplet/src/conn.c:dpl_conn_terminate_nolock:228: conn_terminate conn=0x7fe764207ff0
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [2] ./core/src/droplet/libdroplet/src/conn.c:dpl_conn_free:187: SSL shutdown was successfully completed
Nov 23 12:48:08 server bareos-sd[393924]: trace: 394006: [400] ./core/src/droplet/libdroplet/src/backend/s3/backend/head_raw.c:dpl_s3_head_raw:176: ret=-1
```

**Comment by arogge** [comment]
Hi,
we know libdroplet is inherently broken. I'm working on a replacement that uses `s3cmd` for the up/download (actually, it allows the use of arbitrary external programs/scripts).

If you're feeling lucky you can try the packages from https://download.bareos.org/experimental/PR-1790/ or try to build #1790 yourself.

**Comment by lluuaapp** [comment]
Don't know whether I want to feel lucky ;-)

Did not find any documents on how the configuration will work. Is it really an replacement and I should be able to use the current droplet config?

**Comment by arogge** [comment]
Yes and no.
I haven't written user documentation yet. However, there is a [GIST](https://gist.github.com/arogge/f6334f4c7efc1d2acc3e9b7921048b2f) containing a preliminary version of what it going to be the evaluation guide for that new backend.
I cannot guarantee that this is still 100% accurate, but it is at least something you could try to work with.

Nevertheless, the proper documentation is on the way and will be there before the feature-branch is merged.

**Comment by arogge** [comment]
So that new backend is finally released with Bareos 24.0.0.
Discussion #2063 also contains the evaluation guide, so I'll close this issue.

**Comment by lluuaapp** [comment]
Yes, already using 24.0.0 with `bareos-storage-dplcompat` and have no issues so far. Thanks!
